### PR TITLE
Add example use cases for the proposed permissions model

### PR DESCRIPTION
**Pull request by shardulc**
It's a bit wordy, but it clearly and explicitly resolves any confusion that might exist due to the new permissions model. I haven't talked about teams at all, is it required?

### CODE REVIEW

**Comment by sushain97** [comment]
> I haven't talked about teams at all, is it required?

~~A sentence would be good but I really don't think it matters much.~~

Seems like you had a link, that's probably good enough for now. Maybe make a section stub (i.e. title) and say you'll write it if anyone shows interest.

File: perms-use-cases.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+1. Nub emails `apertium-stuff` about wanting to be an organization member.
+1. Two existing organization members, including Con, reply to that email
+seconding his request.
+1. Pam notes that the two conditions required to become a member as per
+Apertium's [by-laws][2] are satisfied: Nub has contributed code in the past, and he
+has two existing members seconding him. Pam makes Nub an organization member of
+`apertium`.
+
+As an organization member, Nub can:
+ * know who the other members are
```
Comment (sushain97) [inline perms-use-cases.md]: non-public ones

**Comment by sushain97** [comment]
The only improvement I suggest is using real names because it'll be easier to follow. Replace Pam with Fran/Mikel, Mai with Ilnar (?) using apertium-tur or something, Con with yourself or I. I'll merge afterwards and send out the email draft.

**Comment by sushain97** [comment]
Should we mention somewhere that being a member does not implicitly grant you write access to all the repos? This is somewhat unintuitive.

File: perms-use-cases.md [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+Apertium's [by-laws][2] are satisfied: Shardul has contributed code in the past, and he
+has two existing members seconding him. Fran makes Shardul an organization member of
+`apertium`.
+
+As an organization member, Shardul can:
+ * know who the other members are (the ones not marked private)
+ * view, create, talk to, and lead [teams][3]
+ * use [project boards][4] to plan projects
+
+**Note**: Shardul's write access to repositories has *not changed* and so far, he only has write
```
Comment (shardulc) [inline perms-use-cases.md:71]: > Should we mention somewhere that being a member does not implicitly grant you write access to all the repos?

Here. This was in the commit before, too, but seeing that you missed it I gave it a **Note:** at the beginning.

File: perms-use-cases.md [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+access to `apertium/apertium-tur`. For other repositories, he can either follow
+steps 1 and 2 again, or Fran can directly give him write access.
+
+
+### How `apertium/apertium-tur` is released
+
+1. Sushain creates a [draft release][5] in `apertium/apertium-tur` and emails Fran or
+`apertium-stuff` about the planned release.
+1. Fran replies, saying that the release is approved.
+1. Sushain publishes the draft release.
```
Comment (sushain97) [inline perms-use-cases.md:82]: Can we mention that Ilnar can bypass this?

File: perms-use-cases.md [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+`apertium`.
+
+As an organization member, Shardul can:
+ * know who the other members are (the ones not marked private)
+ * view, create, talk to, and lead [teams][3]
+ * use [project boards][4] to plan projects
+
+**Note**: Shardul's write access to repositories has *not changed* and so far, he only has write
+access to `apertium/apertium-tur`. For other repositories, he can either follow
+steps 1 and 2 again, or Fran can directly give him write access.
```
Comment (sushain97) [inline perms-use-cases.md]: or admins? I want it to be clear that the PMC doesn't have to interact all the time

**Comment by shardulc** [comment]
Should we also mention somewhere that Ilnar can make someone else admin without Fran's involvement?

File: perms-use-cases.md (hunk 3 above)
Comment (shardulc) [reply perms-use-cases.md:82]: Ilnar and Sushain both have the permissions to publish releases by themselves, but the approval bit comes from the by-laws <-- is this what we should mention?

**Comment by sushain97** [comment]
Maybe that specific detail belongs in the README? There is a listing of
relevant things admins can do and it might just add clutter here.

**Comment by sushain97** [comment]
Ohh, I see. I suppose we should mention that it's the bylaws. I don't know
if that clause really applies to/is followed by anything except core
releases.

On Feb 19, 2018 1:35 PM, "Shardul Chiplunkar" <notifications@github.com>
wrote:

*@shardulc* commented on this pull request.
------------------------------

In perms-use-cases.md
<https://github.com/sushain97/apertium-on-github/pull/20#discussion_r169148813>
:

> +As an organization member, Shardul can:
+ * know who the other members are (the ones not marked private)
+ * view, create, talk to, and lead [teams][3]
+ * use [project boards][4] to plan projects
+
+**Note**: Shardul's write access to repositories has *not changed*
and so far, he only has write
+access to `apertium/apertium-tur`. For other repositories, he can either follow
+steps 1 and 2 again, or Fran can directly give him write access.
+
+
+### How `apertium/apertium-tur` is released
+
+1. Sushain creates a [draft release][5] in `apertium/apertium-tur`
and emails Fran or
+`apertium-stuff` about the planned release.
+1. Fran replies, saying that the release is approved.
+1. Sushain publishes the draft release.

Ilnar and Sushain both have the permissions to publish releases by
themselves, but the approval bit comes from the by-laws <-- is this what we
should mention?

—
You are receiving this because you commented.

Reply to this email directly, view it on GitHub
<https://github.com/sushain97/apertium-on-github/pull/20#discussion_r169148813>,
or mute the thread
